hugr/src/types.rs

```diff
@@ -448,8 +449,61 @@ pub(crate) fn check_typevar_decl(
 
 #[cfg(test)]
 pub(crate) mod test {
+
     use super::*;
-    use crate::extension::prelude::USIZE_T;
+
+    use crate::{extension::prelude::USIZE_T, ops::AliasDecl};
+
+    use crate::types::TypeBound;
+
+    #[cfg(feature = "proptest")]
+    mod proptest {
+        use crate::ops::AliasDecl;
+        use crate::proptest::TypeDepth;
+        use crate::types::custom::test::proptest::CustomTypeArbitraryParameters;
+        use crate::types::{CustomType, FunctionType, SumType, TypeBound, TypeEnum, TypeRow};
+        use ::proptest::prelude::*;
+        impl Arbitrary for super::TypeEnum {
+            type Parameters = TypeDepth;
```

Comment:
Presumably you've done this because constructing arbitrary instances without enforcing a depth limit leaves to overly-massive/deep structures??

I wonder if an easier way might be just to *probabilistically* instruct the framework to prefer the leaf constructors over those with more nested components. Maybe that doesn't, or is hard to tune to, give sufficient coverage of the nested components!

Reply:
Can you perhaps use the `prop_recursive` thing that appears to have a limit on depth and a target total size? Might that be a much easier way of getting structures of the shape you want?